x/supplier/keeper/supplier_test.go

```diff
@@ -100,22 +108,130 @@ func TestSupplier_GetAll(t *testing.T) {
 	)
 }
 
-func TestSupplier_GetAllSuppliersIterator(t *testing.T) {
+func TestSupplier_GetAllUnstakingSuppliersIterator(t *testing.T) {
 	supplierModuleKeepers, ctx := keepertest.SupplierKeeper(t)
+
+	// Create 6 suppliers with unstaking height
 	suppliers := createNSuppliers(*supplierModuleKeepers.Keeper, ctx, 10)
-	iterator := supplierModuleKeepers.GetAllSuppliersIterator(ctx)
+	for i := 2; i < 8; i++ {
+		suppliers[i].UnstakeSessionEndHeight = 100
+		supplierModuleKeepers.SetSupplier(ctx, suppliers[i])
+	}
+
+	// Get all unstaking suppliers
+	iterator := supplierModuleKeepers.GetAllUnstakingSuppliersIterator(ctx)
 	defer iterator.Close()
 
-	retrievedSuppliers := make([]sharedtypes.Supplier, 0)
+	// Count unstaking suppliers from iterator
+	unstakingCount := 0
 	for ; iterator.Valid(); iterator.Next() {
-		supplier, err := iterator.Value()
-		require.NoError(t, err)
-		retrievedSuppliers = append(retrievedSuppliers, supplier)
+		unstakingCount++
 	}
-	require.ElementsMatch(t,
-		nullify.Fill(suppliers),
-		nullify.Fill(retrievedSuppliers),
+
+	// Verify we found exactly 6 unstaking suppliers
+	require.Equal(t, 6, unstakingCount)
+}
+
+func TestServiceConfigUpdateIterators(t *testing.T) {
```

Comment:
Can you:

- Move the inline constants in this test into variables at the top of the test
- Comment what they are and why
- Do "basic arithmetic" where possible
- Avoid "magic numbers" in line for readability


---

I bet if you give the test w/ those instructions to AI, it'll do it for you ;)